11.0.Dockerfile

```diff
@@ -0,0 +1,152 @@
+FROM debian:9
+
+# Enable Odoo user and filestore
+RUN useradd -md /home/odoo -s /bin/false odoo \
+    && mkdir -p /var/lib/odoo \
+    && chown -R odoo:odoo /var/lib/odoo \
+    && sync
+VOLUME ["/var/lib/odoo"]
+EXPOSE 8069 8072
+
+# Subimage triggers
+ONBUILD ENTRYPOINT ["/opt/odoo/common/entrypoint"]
+ONBUILD CMD ["/usr/local/bin/odoo"]
+ONBUILD ARG AGGREGATE=true
+ONBUILD ARG DEPTH_DEFAULT=1
+ONBUILD ARG DEPTH_MERGE=100
+ONBUILD ARG CLEAN=true
+ONBUILD ARG COMPILE=true
+ONBUILD ARG CONFIG_BUILD=true
+ONBUILD ARG PIP_INSTALL_ODOO=true
+ONBUILD ARG ADMIN_PASSWORD=admin
+ONBUILD ARG SMTP_SERVER=smtp
+ONBUILD ARG PROXY_MODE=false
+ONBUILD ARG WITHOUT_DEMO=all
+ONBUILD ARG PYTHONOPTIMIZE=1
+ONBUILD ENV PYTHONOPTIMIZE="$PYTHONOPTIMIZE"
+ONBUILD ARG PGUSER=odoo
+ONBUILD ARG PGPASSWORD=odoopassword
+ONBUILD ARG PGHOST=db
+ONBUILD ARG PGDATABASE=prod
+# Config variables
+ONBUILD ENV ADMIN_PASSWORD="$ADMIN_PASSWORD" \
+            UNACCENT="$UNACCENT" \
+            PGUSER="$PGUSER" \
+            PGPASSWORD="$PGPASSWORD" \
+            PGHOST="$PGHOST" \
+            PGDATABASE="$PGDATABASE" \
+            PROXY_MODE="$PROXY_MODE" \
+            SMTP_SERVER="$SMTP_SERVER" \
+            WITHOUT_DEMO="$WITHOUT_DEMO"
+ONBUILD ARG LOCAL_CUSTOM_DIR=./custom
+ONBUILD COPY $LOCAL_CUSTOM_DIR /opt/odoo/custom
+# https://docs.python.org/2.7/library/logging.html#levels
+ONBUILD ARG LOG_LEVEL=INFO
+ONBUILD RUN mkdir -p /opt/odoo/custom/ssh \
+            && ln -s /opt/odoo/custom/ssh ~root/.ssh \
+            && chmod -R u=rwX,go= /opt/odoo/custom/ssh \
+            && sync
+ONBUILD RUN ["/opt/odoo/common/build"]
+ONBUILD USER odoo
+
+ARG PYTHONOPTIMIZE=2
+ARG WKHTMLTOPDF_VERSION=0.12.4
+ARG WKHTMLTOPDF_CHECKSUM='049b2cdec9a8254f0ef8ac273afaf54f7e25459a273e27189591edc7d7cf29db'
+ENV DEPTH_DEFAULT=1 \
+    DEPTH_MERGE=100 \
+    EMAIL=https://hub.docker.com/r/tecnativa/odoo \
+    GIT_AUTHOR_NAME=docker-odoo \
+    LC_ALL=C.UTF-8 \
+    OPENERP_SERVER=/opt/odoo/auto/odoo.conf \
+    PATH="~/.local/bin:$PATH" \
+    PUDB_RDB_HOST=0.0.0.0 \
+    PUDB_RDB_PORT=6899 \
+    UNACCENT=true \
+    WAIT_DB=true \
+    WDB_NO_BROWSER_AUTO_OPEN=True \
+    WDB_SOCKET_SERVER=wdb \
+    WDB_WEB_PORT=1984 \
+    WDB_WEB_SERVER=localhost
+
+# Other requirements and recommendations to run Odoo
+# See https://github.com/$ODOO_SOURCE/blob/$ODOO_VERSION/debian/control
+RUN apt-get update \
+    && apt-get -y upgrade \
+    && apt-get install -y --no-install-recommends \
+        python3 ruby-compass \
+        fontconfig libfreetype6 libxml2 libxslt1.1 libjpeg62-turbo zlib1g \
+        libfreetype6 liblcms2-2 libtiff5 tk tcl libpq5 \
+        libldap-2.4-2 libsasl2-2 libx11-6 libxext6 libxrender1 \
+        locales-all zlibc \
+        bzip2 ca-certificates curl gettext-base git gnupg2 nano \
```

Comment:
> gnupg2

Whoah. We gained something GPG in Odoo 11?!

Reply:
No, it's just required to install node + npm below. It seems debian 9 doesn't bundle npm.